hr-data-generator/src/main/java/uk/gov/gchq/palisade/example/hrdatagenerator/CreateDataFile.java

```diff
@@ -42,42 +45,41 @@ public CreateDataFile(final long numberOfEmployees, final long seed, final File
-            Stream<Employee> employeeStream;
-            AvroSerialiser<Employee> employeeAvroSerialiser = new AvroSerialiser<>(Employee.class);
-            // Need one Employee whose manager has a UID of Bob (for examples to work)
-            Employee firstEmployee = Employee.generate(random);
-            Manager[] managers = firstEmployee.getManager();
-            UserId lineManagerUid = managers[0].getUid();
-            lineManagerUid.setId("Eve");
-            managers[0].setUid(lineManagerUid);
-            firstEmployee.setManager(managers);
-            Stream<Employee> firstEmployeeStream = Stream.of(firstEmployee);
-            if (numberOfEmployees > 1) {
-                Stream<Employee> moreEmployeesStream = generateStreamOfEmployees();
-                employeeStream = Stream.concat(firstEmployeeStream, moreEmployeesStream);
-            } else {
-                employeeStream = firstEmployeeStream;
-            }
+        boolean mkdirSuccess = outputFile.getParentFile().mkdirs();
+        if (mkdirSuccess) {
+            try (OutputStream out = new FileOutputStream(outputFile)) {
+                AvroSerialiser<Employee> employeeAvroSerialiser = new AvroSerialiser<>(Employee.class);
+
+                // Need one Employee whose manager has a UID of Bob (for examples to work)
+                Employee firstEmployee = Employee.generate(random);
+                Manager[] managers = firstEmployee.getManager();
+                UserId lineManagerUid = managers[0].getUid();
+                lineManagerUid.setId("Eve");
+                managers[0].setUid(lineManagerUid);
+                firstEmployee.setManager(managers);
 
-            employeeAvroSerialiser.serialise(employeeStream, out);
-        } catch (final Exception error) {
-            error.printStackTrace();
+                // Create more employees if needed
+                Stream<Employee> employeeStream = Stream.of(firstEmployee);
+                if (numberOfEmployees > 1) {
+                    employeeStream = Stream.concat(employeeStream, generateStreamOfEmployees());
+                }
+
+                // Serialise stream to output
+                employeeAvroSerialiser.serialise(employeeStream, out);
+                return true;
+            } catch (Exception ex) {
+                LOGGER.error("", ex);
+                return false;
+            }
+        } else {
+            return false;
         }
-        return Boolean.TRUE;
     }
 
     private Stream<Employee> generateStreamOfEmployees() {
-        final AtomicLong counter = new AtomicLong(0);
-        final long countWrite = numberOfEmployees / 10;
-        return Stream.generate(() -> {
-            long count = counter.incrementAndGet();
-            if ((numberOfEmployees > 1_000_000 && countWrite > 0 && count % countWrite == 0) || count % 100_000 == 0) {
-                System.err.printf("Thread %s has written %d records.%n", Thread.currentThread().getName(), count);
-            }
```

Comment:
So this section was there to provide a stream of logs every 100_000 employee's generated if you are generating more than a million employees so you know it is still processing as intended rather than having frozen or hit an infinite loop. This is important when generating very large volumes of test data